syncer/dml.go

```diff
@@ -140,16 +177,30 @@ func genUpdateSQLs(schema string, table string, data [][]interface{}, columns []
 			changedValues = append(changedValues, castUnsigned(changedData[i], columns[i].unsigned, columns[i].tp))
 		}
 
+		oriOldValues := make([]interface{}, 0, len(oriOldData))
```

Comment:
`genUpdateSQLs`, `genInsertSQLs`` has these codes in same pattern:

```
...values
for i := range data {
    values = append(values,....)
}

if len(columns) == len(originalColumns) {
  originalVals = values
} else {
    for {....}
}
```

can we extract it a function?